tests/integration/interactive/cmd_init_test.go

```diff
@@ -59,4 +61,126 @@ var _ = Describe("odo init interactive command tests", func() {
 		Expect(output).To(ContainSubstring("Your new component \"my-go-app\" is ready in the current directory."))
 		Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElements("devfile.yaml"))
 	})
+
+	Describe("Displaying welcoming messages", func() {
+
+		type tester func(*expect.Console, *bytes.Buffer)
+
+		testFunc := func(language string, welcomingMsgs []string, additionalTester tester) tester {
+			return func(c *expect.Console, output *bytes.Buffer) {
+				var res string
+				for _, msg := range welcomingMsgs {
+					res = helper.ExpectString(c, msg)
+					fmt.Fprint(output, res)
+				}
+
+				if additionalTester != nil {
+					additionalTester(c, output)
+				}
+
+				res = helper.ExpectString(c, "Enter component name")
+				fmt.Fprintln(output, res)
+				helper.SendLine(c, fmt.Sprintf("my-%s-app", language))
+
+				res = helper.ExpectString(c,
+					fmt.Sprintf("Your new component \"my-%s-app\" is ready in the current directory.", language))
+				fmt.Fprintln(output, res)
+			}
+		}
+
+		assertBehavior := func(language string, output string, err error, msgs []string, additionalAsserter func()) {
+			Expect(err).To(BeNil())
+			Expect(output).To(SatisfyAll(
+				HavePrefix(strings.Join(msgs, "\r\n")),
+				ContainSubstring(
+					fmt.Sprintf("Your new component \"my-%s-app\" is ready in the current directory.", language))))
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElements("devfile.yaml"))
+			if additionalAsserter != nil {
+				additionalAsserter()
+			}
+		}
+
+		When("directory is empty", func() {
+			It("should display appropriate welcoming messages", func() {
+				command := []string{"odo", "init"}
+				language := "java"
+				welcomingMsgs := []string{"The current directory is empty.", "Odo will help with you starting a new project."}
+				output, err := helper.RunInteractive(command,
+					// Setting verbosity level to 0, because we would be asserting the welcoming message is the first
+					// message displayed to the end user. So we do not want any potential debug lines to be printed first.
+					// Using envvars here (and not via the -v flag), because of https://github.com/redhat-developer/odo/issues/5513
+					[]string{"ODO_LOG_LEVEL=0"},
+					testFunc(language, welcomingMsgs, func(c *expect.Console, output *bytes.Buffer) {
```

Comment:
As this block of test is already used in the previous tests, it would be nice to define this function at a higher level so it can be reused in different tests

Reply:
Done. Please take another look.